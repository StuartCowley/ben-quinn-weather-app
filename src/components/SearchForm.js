import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => {
    setSearchText(event.target.value);
  };

  return (
    <div className="search-form">
      <input
        className="search-form__input"
        data-testid="search-form__input"
        type="text"
        placeholder="Enter city..."
        onChange={handleInputChange}
        value={searchText}
      />
      <button
        className="search-form__button"
        data-testid="search-form__button"
        type="submit"
        onClick={onSubmit}
      >
        <span className="material-icons" data-testid="search-form__icon">
          search
        </span>
      </button>
    </div>
  );
}

export default SearchForm;

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
