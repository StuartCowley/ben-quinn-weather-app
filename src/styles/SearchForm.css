.search-form {
  display: flex;
  justify-content: center;
  margin-top: 10px;
  margin-bottom: 10px;
}

input[type="text"] {
  background: transparent;
  border: 0;
  border-bottom: 3px solid black;
  width: 200px;
}

input[type="text"]:focus {
  outline: none;
}

.search-form__button {
  margin-left: 10px;
  border: 2px;
  background: transparent;
}
