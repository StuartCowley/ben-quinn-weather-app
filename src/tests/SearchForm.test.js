import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "searchText",
    setSearchText: () => {},
    handleCitySearch: () => {},
  };

  it("Renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
        searchText={validProps.searchText}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders button with correct text", () => {
    const { getByText } = render(
      <SearchForm
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
        searchText={validProps.searchText}
      />
    );
    const button = getByText(/search/i);

    expect(button).toHaveClass("search-form__button");
    expect(button).toHaveAttribute("type", "submit");
  });

  it("Renders input element", () => {
    const { getByTestId } = render(
      <SearchForm
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
        searchText={validProps.searchText}
      />
    );
    const input = getByTestId("search-form__input");

    expect(input).toHaveAttribute("type", "text");
  });
});
