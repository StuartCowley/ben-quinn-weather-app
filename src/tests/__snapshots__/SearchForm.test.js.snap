// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-form__input"
      data-testid="search-form__input"
      placeholder="Enter city..."
      type="text"
      value="searchText"
    />
    <button
      class="search-form__button"
      data-testid="search-form__button"
      type="submit"
    >
      <span
        class="material-icons"
        data-testid="search-form__icon"
      >
        search
      </span>
    </button>
  </div>
</DocumentFragment>
`;
